hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceImpl.java

```diff
@@ -491,7 +491,7 @@ private SerializerAdapter createSerializerAdapter(Serializer serializer) {
 
     protected final SerializerAdapter serializerFor(final Class type) {
         SerializerAdapter serializer;
-        if (overrideCustomSerialization) {
+        if (overrideCustomSerialization && constantTypesMap.get(type) == null) {
```

Comment:
This is there for to prevent constant types are not picked up by the custom serializer when serialization override is enabled. This logic will be changed by the @asimarslan work on serialization.